app/routes.py

```diff
@@ -1 +1,286 @@
-from flask import Blueprint
\ No newline at end of file
+
+from app.models.task import Task
+from app.models.goal import Goal
+from app import db
+from flask import Blueprint, jsonify,abort,make_response,request
+from sqlalchemy.sql.functions import now
+import os
+import requests
+
+
+task_list_bp = Blueprint("task_list", __name__,url_prefix="/tasks")
+goal_list_bp = Blueprint("goal_list", __name__,url_prefix="/goals")
+
+
+def validate_task(task_id):
+    try:
+       task_id = int(task_id)
+    except:
+        abort(make_response({"message":f"task {task_id} invalid"}, 400))
+
+    task = Task.query.get(task_id)
+
+    if not task:
+        abort(make_response({"message":f"task {task_id} not found"}, 404))
+
+    return task
+
+
+
+
+#Get tasks sorted or get all tasks
+@task_list_bp.route("", methods=["GET"])
+def get_tasks_sorted():
+   
+    sort_query = request.args.get("sort")
+
+   
+    if sort_query and sort_query=="asc":
+        tasks= Task.query.order_by(Task.title.asc())
+    elif sort_query and sort_query=="desc":
+        tasks= Task.query.order_by(Task.title.desc())
+    else:
+        tasks = Task.query.all()
+
+
+        
+    task_response = []
+    for task in tasks:
+        task_response.append(task.to_json())
+    return make_response( jsonify(task_response),200)
+
+
+
+
+#Get one task
+@task_list_bp.route("/<task_id>", methods=["GET"])
+def get_one_task(task_id):
+   
+    task = Task.query.get(task_id)
+
+    if task:
+        return {
+            "task": task.to_json()
+        }
```

Comment:
this works, but don't forget your status code! every response we return to our client should be consistent and predictable

```suggestion
        return jsonify({"task": task.to_json()}), 200
```